File: examples/99.ex
for n <- 99..1 |> Enum.reverse do
  IO.puts("#{n} bottle#{if n==1, do: "", else: "s"} of beer on the wall, #{n} bottle#{if n==1, do: "", else: "s"} of beer.")
  if n-1 > 0 do
    IO.puts("Take one down and pass it around, #{n-1} bottle#{if n-1==1, do: "", else: "s"} of beer on the wall.\n")
  else
    IO.puts("Take one down and pass it around, no more bottles of beer on the wall.\n")
  end
end
IO.puts("No more bottles of beer on the wall, no more bottles of beer.")
IO.puts("Go to the store and buy some more, 99 bottles of beer on the wall.")

File: examples/99.cpp
#include <iostream>
using namespace std;
int main(){
  for(int n=99;n>0;--n){
    cout<<n<<" bottle"<<(n==1?"":"s")<<" of beer on the wall, "<<n<<" bottle"<<(n==1?"":"s")<<" of beer.\n";
    if(n-1>0) cout<<"Take one down and pass it around, "<<n-1<<" bottle"<<(n-1==1?"":"s")<<" of beer on the wall.\n\n";
    else cout<<"Take one down and pass it around, no more bottles of beer on the wall.\n\n";
  }
  cout<<"No more bottles of beer on the wall, no more bottles of beer.\n";
  cout<<"Go to the store and buy some more, 99 bottles of beer on the wall.\n";
}
